### PR TITLE
Documentation: Fix broken links, move deep dive section

### LINKED ISSUE #2305: [DOCS] Broken links in readme
The links in tracee-ebpf and tracee-rules readme are broken:
1. https://github.com/aquasecurity/tracee/blob/main/cmd/tracee-ebpf/readme.md
2. https://github.com/aquasecurity/tracee/blob/main/cmd/tracee-rules/readme.md

These should point to the tracing and detecting pages in the docs respectively.

In addition, the components part in the main readme should be updated as well. https://github.com/aquasecurity/tracee/blob/main/Readme.md#components 
Currently it points to the commands pages, but the readme in those pages simply points to the docs, so we can just directly point to the docs instead of this extra indirection

### PR DESCRIPTION
**Pull request by grantseltzer**
## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [ ] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

commit d0bf43115316c65bb9fb51b74d946acdc476e017 (HEAD -> docs-changes, pfork/docs-changes)
Author: grantseltzer <grantseltzer@gmail.com>
Date:   Mon Oct 31 14:04:34 2022 -0400

    Link to components docs directly
    
    Signed-off-by: grantseltzer <grantseltzer@gmail.com>

commit fc7634950e004455b94baf427479a7004fb717ef
Author: grantseltzer <grantseltzer@gmail.com>
Date:   Mon Oct 31 14:01:26 2022 -0400

    Fix broken links
    
    Signed-off-by: grantseltzer <grantseltzer@gmail.com>

commit 6bfc0a6cb917c0c53e440b31a31c2d473981b030
Author: grantseltzer <grantseltzer@gmail.com>
Date:   Mon Oct 31 13:47:05 2022 -0400

    Move Deep Dive to docs rather than Contributing section
    
    Signed-off-by: grantseltzer <grantseltzer@gmail.com>

Fixes #2305
## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [ ] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?

```
make -f builder/Makefile.mkdocs mkdocs-build
make -f builder/Makefile.mkdocs mkdocs-serve
```

### CODE REVIEW

File: docs/contributing/overview.md [hunk 1]
```diff
@@ -1 +1,53 @@
...
+### Discussions:
+
+1. We have the following discussion topics: 
+  1. [Announcements](https://github.com/aquasecurity/tracee/discussions/categories/announcements): One way communication from the team to the community. Consider this like our mini blog
+  2. [Questions and Help](https://github.com/aquasecurity/tracee/discussions/categories/questions-and-help): For help and support. Consider this similar to StackOverflow.
+  3. [Development](https://github.com/aquasecurity/tracee/discussions/categories/development): For discussing potential features, and collaborating on their design.
+
+### Issues:
+
+1. Every issue needs to be actionable and assignable. Consider the scope of the issue if assigned to one person, and break down if necessary.
```
Comment (AnaisUrlichs) [inline docs/contributing/overview.md:23]: the points here don't seem to be working properly? i.e. 
1. ....
1. ....
     1.
     2.

File: docs/contributing/overview.md [hunk 2]
```diff
@@ -1 +1,53 @@
-This section goes deep explaning Tracee's internals, and how to build Tracee on different platforms for the users interested in contributing.
+## Contributing
```
Comment (AnaisUrlichs) [inline docs/contributing/overview.md:1]: Once merged, you should maybe link to this file from the PR/issue templates? cc @grantseltzer

**Review by josedonizetti — COMMENTED** [review]
Can we add a section with overview about the new sections under Docs? Eg: https://aquasecurity.github.io/tracee/v0.9/docs/overview/

File: docs/contributing/overview.md (hunk 1 above)
Comment (rafaeldtinoco) [reply docs/contributing/overview.md:23]: I haven't checked but I would like to raise awareness for: bullets/numbers for markdown are more complete than from mkdocs. I had to make sure that both worked when I did the initial documentation (and the end of the lines, for mkdocs, had to have 2 spaces "  " for line breakage). So there are some caveats and we have to make sure both are good: markdown and mkdocs. My 5 cents only.

Comment (grantseltzer) [reply docs/contributing/overview.md:23]: Good call out. I fixed the issue here (and elsewhere in the file) and verified that it rendered well on both github and in mkdocs. Interestingly, the sublists render as numbers on github and letters in mkdocs. Either way looks good.

File: docs/contributing/overview.md (hunk 2 above)
Comment (grantseltzer) [reply docs/contributing/overview.md:1]: Will do!

**Review by josedonizetti — APPROVED** [review]

**Pull request merged**